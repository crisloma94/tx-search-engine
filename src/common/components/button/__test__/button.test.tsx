import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Theme from '../../../../styles/theme';
import { store } from '../../../../core/store/store';
import Button, { ButtonProps } from '../button.component';

describe('Button component test suite', () => {
  const getComponent = (props: ButtonProps) => (
    <Provider store={store}>
      <Theme>
        <Button {...props} />
      </Theme>
    </Provider>
  );
  const setup = (props: ButtonProps) => render(getComponent(props));

  const onClickMock = jest.fn();
  const defaultProps = { text: 'Button text', onClick: onClickMock };

  it('Should match snapshot', () => {
    const { container } = setup(defaultProps);
    expect(container).toMatchSnapshot();
  });

  it('Should call the onClick function when clicked', async () => {
    setup(defaultProps);
    fireEvent.click(screen.getByRole('button'));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
