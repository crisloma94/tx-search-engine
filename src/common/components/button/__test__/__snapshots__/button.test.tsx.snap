// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test suite Should match snapshot 1`] = `
<div>
  <button
    class="sc-bgqQcB iqnJhc"
    type="button"
  >
    Button text
  </button>
</div>
`;
